extract ignore_order from kwargs

galipremsagar opened the pull request:
When https://github.com/dask/dask/pull/7473 was merged, `ignore_order` was added in the middle of the function signature of `concat_dispatch`, this leads to breakages in downstream libraries using this dispatch. Hence with these changes `ignore_order` will be extracted from `**kwargs`. 

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask` / `isort dask`

jakirkham commented:
cc @mesejo

jakirkham commented on dask/dataframe/methods.py:
```diff
@@ -426,6 +425,7 @@ def concat(
     if len(dfs) == 1:
         return dfs[0]
     else:
+        ignore_order = kwargs.get("ignore_order", False)
```
Would suggest using `pop` as we are passing this and `**kwargs` later

```suggestion
        ignore_order = kwargs.pop("ignore_order", False)
```

jakirkham commented on dask/dataframe/methods.py:
```diff
@@ -446,13 +446,14 @@ def concat_pandas(
     if axis == 1:
         return pd.concat(dfs, axis=axis, join=join, **kwargs)
 
+    ignore_order = kwargs.get("ignore_order", False)
```
Would also suggest using `pop` here as `**kwargs` is getting forwarded to Pandas, which doesn't know how to handle this (causing CI failures atm)

```suggestion
    ignore_order = kwargs.pop("ignore_order", False)
```

mesejo commented:
Hi, I think this will break usage of concat in the function `stack_partitions` (line 1078):

`                dsk[(name, i)] = (
                    methods.concat,
                    [empty, key],
                    0,
                    join,
                    uniform,
                    filter_warning,
                    ignore_order,
                )`  

where is expecting `ignore_order` as a regular parameter. I believe changing the order of the parameters and providing the default value for `ignore_index` should be enough.

The same happens in the function `concat_indexed_dataframes` (line 1009)

mesejo commented:
As a side note, the parameter `ignore_index` is not being used inside the function `concat_pandas` (line 443, methods.py)

galipremsagar commented:
@mesejo I have fixed the callers.

jakirkham commented:
Thanks for taking a look Daniel! 😄

> As a side note, the parameter `ignore_index` is not being used inside the function `concat_pandas` (line 443, methods.py)

Interesting so to help us understand this better. Was the idea to eventually make use of this parameter there as well? Or was it just to match the higher level `concat` function's signature? Asking since this argument was added to `concat_pandas` prior

mesejo commented:
> @mesejo I have fixed the callers.

Great! Please also change this line (1009, multi.py) inside `concat_indexed_dataframes` is similar to the one in `stack_partitions`:

`            (methods.concat, part, axis, join, uniform, filter_warning, ignore_order),`

mesejo commented:
> Thanks for taking a look Daniel! smile
> 
> > As a side note, the parameter `ignore_index` is not being used inside the function `concat_pandas` (line 443, methods.py)
> 
> Interesting so to help us understand this better. Was the idea to eventually make use of this parameter there as well? Or was it just to match the higher level `concat` function's signature? Asking since this argument was added to `concat_pandas` prior

Glad I could help! To be honest I don't know as you said it was already there. I just mentioned it because if some downstream library is relying on this to make a change in the behavior of the function it won't work.

galipremsagar commented:
> > @mesejo I have fixed the callers.
> 
> Great! Please also change this line (1009, multi.py) inside `concat_indexed_dataframes` is similar to the one in `stack_partitions`:
> 
> ` (methods.concat, part, axis, join, uniform, filter_warning, ignore_order),`

Thanks @mesejo, fixed this one.

jakirkham commented on dask/dataframe/multi.py:
```diff
@@ -1081,7 +1080,8 @@ def stack_partitions(dfs, divisions, join="outer", ignore_order=False, **kwargs)
                     filter_warning,
-                    ignore_order,
+                    False,
+                    kwargs,
```
It looks like we are adding `False` here. Sorry if I missed this somewhere, but could you please remind me why that is?

galipremsagar replied:
Ah I missed this one, added it for testing locally. Thanks for this catch John.

galipremsagar commented on dask/dataframe/multi.py:
```diff
@@ -1081,7 +1080,8 @@ def stack_partitions(dfs, divisions, join="outer", ignore_order=False, **kwargs)
                     uniform,
                     filter_warning,
-                    ignore_order,
+                    False,
+                    kwargs,
```
```suggestion
                    kwargs,
```

jakirkham commented on dask/dataframe/multi.py:
```diff
@@ -1081,7 +1080,7 @@ def stack_partitions(dfs, divisions, join="outer", ignore_order=False, **kwargs)
                     uniform,
                     filter_warning,
-                    ignore_order,
+                    kwargs,
```
```suggestion
                    **kwargs,
```

jakirkham commented:
@jsignell would you be able to take a look at this? 🙂

jsignell reviewed (approved):
This looks reasonable to me. Thanks for writing it up!
